readthedocs/projects/models.py

```diff
@@ -707,7 +707,7 @@ def update_stable_version(self):
             if current_stable:
                 identifier_updated = (
                     new_stable.identifier != current_stable.identifier)
-                if identifier_updated and current_stable.active:
+                if identifier_updated and current_stable.active and current_stable.machine:
```

Comment:
So, comparing this with the original code, we are just adding a another condition (has to be `active`) but we keep the `machine`.

In other words, we are just updating the stable version only if, the identifier changed (a new tag become the stable, for example) and the stable version is active and it was created by RTD.

If what I'm saying is correct, I think we are OK :)

Reply:
Yep, thats the goal :)